api/v2/api.go

```diff
@@ -315,6 +315,132 @@ func (api *API) ListenAndServe(ctx context.Context, addr string, tlsConfig *TLSC
+			api.queues.dash = qmDash
+		case msg := <-api.queues.database.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "database-file-add",
+				"error", msg.Error())
+			qmDatabase, err := queue.New(queue.DatabaseFileAddQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "database-file-add",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.database = qmDatabase
+		case msg := <-api.queues.email.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "email-send",
+				"error", msg.Error())
+			qmEmail, err := queue.New(queue.EmailSendQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "email-send",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.email = qmEmail
+		case msg := <-api.queues.file.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "ipfs-file",
+				"error", msg.Error())
+			qmFile, err := queue.New(queue.IpfsFileQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "ipfs-file",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.file = qmFile
+		case msg := <-api.queues.ipns.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "ipns-entry",
+				"error", msg.Error())
+			qmIpns, err := queue.New(queue.IpnsEntryQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "ipns-entry",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.ipns = qmIpns
+		case msg := <-api.queues.key.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "key-creation",
+				"error", msg.Error())
+			qmKey, err := queue.New(queue.IpfsKeyCreationQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "key-creation",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.key = qmKey
+		case msg := <-api.queues.payConfirm.ErrChannel:
+			api.l.Errorw(
+				"a protocol connection error stopping rabbitmq was received",
+				"queue", "pay-confirm",
+				"error", msg.Error())
+			qmPay, err := queue.New(queue.PaymentConfirmationQueue, api.cfg.RabbitMQ.URL, true, api.l)
+			if err != nil {
+				api.l.Errorw(
+					"failed to re-establish queue process, exiting",
+					"queue", "pay-confirm",
+					"error", err.Error())
+				return server.Close()
+			}
+			api.queues.payConfirm = qmPay
```

Comment:
a lot of repetition here - maybe something like:

```go
// this doesnt exist yet but would be nice to have
type Queue string

// this doesnt exist yet but would be nice to have
type queue.Options{
   mqUrl string
   publish bool
   logger *zap.Logger
}

func handleQueueError(err error, m *queue.Manager, q queue.Queue, opts queue.Options) error {
	api.l.Errorw("a protocol connection error stopping rabbitmq was received",
				"queue", q,
				"error", msg.Error())
         manager, err = queue.New(q, opts) // assigns to appropriate queue
        return err
}
```

Reply:
very nice will implement :ok_hand:

Reply:
Ok I implemented like 75% of this suggestion. I think we might be able to do something with our `API::Queues` struct, and have each individual queue be a `QueueSettings` or something but i'll link into this and if it is possible, do it in a separate PR.